### PR TITLE
Add interval to statsV2

### PR DESCRIPTION
**Pull request by vanchaxy**
This pull request adds an "Interval" field to the stats query, allowing users to customize data grouping for more precise monitoring and analysis.
image

### CODE REVIEW

**Review by aangelisc — APPROVED** [review]
Hi @vanchaxy, thank you for the contribution! This LGTM just one nit in comments

File: pkg/plugin/handlers_query.go [hunk 1]
```diff
@@ -85,6 +86,7 @@ func QueryData(ctx context.Context, pCtx backend.PluginContext, backendQuery bac
 			Category:         query.StatsCategory,
 			Fields:           query.StatsFields,
 			GroupBy:          query.StatsGroupBy,
+			Interval:         query.StatsInterval,
```
Comment (aangelisc) [inline pkg/plugin/handlers_query.go:89]: Could you add one test in `handlers_query_test.go` for this case?

File: src/selectors.ts [hunk 2]
```diff
@@ -79,6 +79,11 @@ export const Components = {
         tooltip: 'group by',
         placeholder: 'Optional. Select the option to group the results',
       },
+      Interval: {
+        label: 'Interval',
+        tooltip: 'interval',
+        placeholder: 'Optional. Enter the interval to group the results',
```
Comment (aangelisc) [inline src/selectors.ts:85]: Could we expand on this by giving details on the expected string e.g.
```suggestion
        placeholder: 'Optional. Enter the interval to group the results (expected format `<number><unit>` where unit is `m` for minutes, `h` for hours, `d` for days, or `w` for weeks)',
```

File: pkg/plugin/handlers_query.go (hunk 1 above)
Comment (vanchaxy) [reply pkg/plugin/handlers_query.go:89]: I added regex validation that the interval format is correct and two tests for correct and incorrect formats.

**Comment by vanchaxy** [comment]
Can someone tell me why the Drone is failing and how can I check it or run CI locally, as the Drone instance isn't public?

**Comment by aangelisc** [comment]
Hi @vanchaxy, Drone is failing due to a spellcheck issue in `pkg/sentry/stats_v2.go`. The word `mhdw` needs to be defined in the `cspell.config.json` and then this should pass 😊

**Review by aangelisc — APPROVED** [review]

**Comment by vanchaxy** [comment]
Hi @aangelisc , thanks for the review. Added "mhdw" to the `cspell.config.json` file but Drone still fails.

**Comment by aangelisc** [comment]
Hey @vanchaxy, you now have a failing backend lint I believe:
image

**Pull request merged**